docs/custom-filters.adoc

```diff
@@ -36,7 +47,100 @@ For common things like logging and metric facade APIs it is recommended to use t
 == Protocol filters
 
 A protocol filter is a `public` top-level, concrete class with a particular public constructor and which implements
-one or more protocol filter interfaces.
+one or more protocol filter interfaces. You can implement two distinct types of Custom Protocol Filter:
+
+- <<Specific Message Protocol Filters>>
+- <<Request/Response Protocol Filters>>
+
+Note that these types are mutually exclusive, for example a Filter is not allowed to implement RequestFilter and
+MetadataRequestFilter.
+
+You can also implement a convenience interface to deliver multiple Protocol Filters to the filter chain using <<Composite Filters>>
+
+=== Specific Message Protocol Filters
+
+A filter may wish to intercept specific types of Kafka messages. For example, intercept all Produce Requests, or
+intercept all Fetch Responses. To support this case Kroxylicious provides an interfaces for all request types and
+response types supported by Kafka (at the version of Kafka Kroxylicious depends on).
```

Comment:
I think its worth making it explicit that there is no requirement for Filters to be symmetric. i.e. If they implement `ProduceRequestFilter` they don't need to implement `ProduceResponseFilter`. Though many will do both.